roles/grafana_cloud_operator/tasks/grafana_oncall_hub_spoke.yml

```diff
@@ -11,19 +11,39 @@
       metadata:
         name: "{{ cr_name }}"
         namespace: "{{ cr_namespace }}"
-      status:
-        lastUpdated: "{{ ansible_date_time.iso8601 }}"
-        phase: "Initiated"
-        reason: "OperationStarted"
-        message: "Operation to integrate Grafana OnCall started"
+        finalizers:
+          - "grafanacloud.stakater.com/config"
+  when: "'grafanacloud.stakater.com/config' not in gco_cr.metadata.finalizers"
+
+- name: Update CR status to Initiated
+  operator_sdk.util.k8s_status:
+    api_version: grafanacloud.stakater.com/v1alpha1
+    kind: Config
+    name: "{{ cr_name }}"
+    namespace: "{{ cr_namespace }}"
+    status:
+      conditions:
+        - lastTransitionTime: "{{ ansible_date_time.iso8601 }}"
+          status: 'True'
+          type: "Initiated"
+          reason: "OperationStarted"
+          message: "Operation to integrate Grafana OnCall started"
+
+- name: Fetch current status from Config CR
+  kubernetes.core.k8s_info:
+    api_version: grafanacloud.stakater.com/v1alpha1
+    kind: Config
+    name: "{{ cr_name }}"
+    namespace: "{{ cr_namespace }}"
+  register: current_config_cr
 
 - name: Fetch ManagedCluster from all namespaces
   kubernetes.core.k8s_info:
     api_version: cluster.open-cluster-management.io/v1
     kind: ManagedCluster
   register: managed_clusters_raw
 
-- name: Extract list of ManagedCluster with their namespaces
+- name: Extract list of ManagedCluster and use name as key
```

Comment:
Extract list of ManagedCluster and use name as key => 

Extract the list of ManagedCluster CRs

Reply:
In the next step: 

Determine which ManagedCluster don't have integrations
=> 
Determine which ManagedCluster CRs don't have integrations